src/main/java/org/apache/mesos/curator/CuratorConfigStore.java

```diff
@@ -41,30 +49,50 @@
     private final String targetPath;
 
     /**
+     * Creates a new {@link ConfigStore} which uses Curator with a default {@link RetryPolicy} and
+     * connection string.
+     *
+     * @param frameworkName The name of the framework
+     */
+    public CuratorConfigStore(String frameworkName) {
+        this(frameworkName, CuratorUtils.DEFAULT_CONNECTION_STRING);
+    }
+
+    /**
      * Creates a new {@link ConfigStore} which uses Curator with a default {@link RetryPolicy}.
      *
-     * @param rootPath The path to store data in, eg "/FrameworkName"
+     * @param frameworkName The name of the framework
      * @param connectionString The host/port of the ZK server, eg "master.mesos:2181"
      */
-    public CuratorConfigStore(String rootPath, String connectionString) {
-        this(rootPath, connectionString, new ExponentialBackoffRetry(
-                DEFAULT_CURATOR_POLL_DELAY_MS, DEFAULT_CURATOR_MAX_RETRIES));
+    public CuratorConfigStore(String frameworkName, String connectionString) {
+        this(frameworkName, connectionString, new ExponentialBackoffRetry(
+                CuratorUtils.DEFAULT_CURATOR_POLL_DELAY_MS,
+                CuratorUtils.DEFAULT_CURATOR_MAX_RETRIES));
     }
 
     /**
      * Creates a new {@link ConfigStore} which uses Curator with a custom {@link RetryPolicy}.
      *
-     * @param rootPath The path to store data in, eg "/FrameworkName"
+     * @param frameworkName The name of the framework
      * @param connectionString The host/port of the ZK server, eg "master.mesos:2181"
      * @param retryPolicy The custom {@link RetryPolicy}
      */
-    public CuratorConfigStore(String rootPath, String connectionString, RetryPolicy retryPolicy) {
-        if (!rootPath.startsWith("/")) {
-            throw new IllegalArgumentException("rootPath must start with '/': " + rootPath);
-        }
+    public CuratorConfigStore(
+            String frameworkName, String connectionString, RetryPolicy retryPolicy) {
         this.curator = new CuratorPersister(connectionString, retryPolicy);
-        this.targetPath = rootPath + "/" + TARGET_PATH_NAME;
-        this.configurationsPath = rootPath + "/" + CONFIGURATIONS_PATH_NAME;
+        String rootPath = CuratorUtils.toServiceRootPath(frameworkName);
+
+        // Check version up-front:
+        SchemaVersionStore versionStore = new CuratorSchemaVersionStore(curator, rootPath);
+        int currentVersion = versionStore.fetch();
+        if (!SchemaVersionStore.isSupported(
+                currentVersion, MIN_SUPPORTED_SCHEMA_VERSION, MAX_SUPPORTED_SCHEMA_VERSION)) {
+            throw new IllegalStateException(String.format(
+                    "Storage schema version %d is not supported by this software", currentVersion));
```

Comment:
Might be nice to include the MIN and MAX supported versions here in the exception, for logging/debugging purposes.